### ISSUE TITLE
Can i get top-5 precision , recall , f1 scores ?

### ISSUE THREAD

**Issue by Balakumaran-kandula** [open-mmlab/mmclassification#847]
We have the feature for top 5 accuracy , can i get precision , recall , f1 scores for top 5 , top 10 also.
And also can i get roc , auc and specificity ?

**Comment by Ezra-Yu** [comment]
I'm in the process of refactoring the metrics. so stay tuned.

**Comment by Xglbrilliant** [comment]
> I'm in the process of refactoring the metrics. so stay tuned.

Hello, may I ask if the indicators of auc and roc can be put into use?I am looking forward to your work.

**Comment by ck37** [comment]
We're also hoping to be able to calculate AUC for a single label - maybe we can develop a PR though if one isn't already in the works.

**Comment by fansofml** [comment]
After two days debug and define metric,now I have a easy solution for two classifier...
you can redefine the Accuary, and use the sklearn to do it.

`
from sklearn.metrics import roc_auc_score,roc_curve,auc
#--------------------------this is the origin code------------------------------------------

        pred = to_tensor(pred)
        target = to_tensor(target).to(torch.int64)
        num = pred.size(0)
        assert pred.size(0) == target.size(0), \
            f"The size of pred ({pred.size(0)}) doesn't match "\
            f'the target ({target.size(0)}).'
#--------------------------------this is my code----------------------------------------------
# add this code in the    calculate() of Accuracy function


        fpr, tpr, thresholds = roc_curve(target, [p[1] for p in pred])

        # 计算ROC曲线下面积（AUC）
        roc_auc = auc(fpr, tpr)
        print("AUC:",roc_auc )

`
